rhine-x11/src/FRP/Rhine/X11.hs

```diff
@@ -0,0 +1,42 @@
+module FRP.Rhine.X11 where
+
+import           Control.Monad             (when)
+import           Data.Functor              (void)
+import           Data.Maybe                (fromMaybe, isJust)
+import           Data.Time.Clock           (getCurrentTime)
+import           FRP.Rhine
+import           Graphics.X11              (allocaSetWindowAttributes,
+                                            cWEventMask, propertyChangeMask,
+                                            set_event_mask)
+import           Graphics.X11.Xlib         (nextEvent, openDisplay)
+import           Graphics.X11.Xlib.Display (defaultRootWindow)
+import           Graphics.X11.Xlib.Event   (allocaXEvent)
+import           Graphics.X11.Xlib.Extras  (Event, changeWindowAttributes,
+                                            getEvent)
+import           Graphics.X11.Xrandr       (xrrQueryExtension,
+                                            xrrUpdateConfiguration)
+import           Prelude
+
+newtype X11Clock = X11Clock{displayName :: Maybe String}
+
+instance MonadIO m => Clock m X11Clock where
+    type Time X11Clock = UTCTime
+    type Tag X11Clock = Event
+    initClock :: X11Clock -> RunningClockInit m (Time X11Clock) (Tag X11Clock)
+    initClock X11Clock{..} = liftIO do
+        display <- openDisplay $ fromMaybe "" displayName
```

Comment:
Should this be `fromMaybe "DISPLAY"`?

Reply:
Nope, it appears to want an empty string.